src/runtime/ltypes/goto.py

```diff
@@ -0,0 +1,235 @@
+import dis
+import struct
+import array
+import types
+import functools
+
+
+try:
+    _array_to_bytes = array.array.tobytes
+except AttributeError:
+    _array_to_bytes = array.array.tostring
+
+
+class _Bytecode:
+    def __init__(self):
+        code = (lambda: x if x else y).__code__.co_code
+        opcode, oparg = struct.unpack_from('BB', code, 2)
+
+        # Starting with Python 3.6, the bytecode format has changed, using
+        # 16-bit words (8-bit opcode + 8-bit argument) for each instruction,
+        # as opposed to previously 24 bit (8-bit opcode + 16-bit argument)
+        # for instructions that expect an argument and otherwise 8 bit.
+        # https://bugs.python.org/issue26647
+        if dis.opname[opcode] == 'POP_JUMP_IF_FALSE':
+            self.argument = struct.Struct('B')
+            self.have_argument = 0
+            # As of Python 3.6, jump targets are still addressed by their
+            # byte unit. This is matter to change, so that jump targets,
+            # in the future might refer to code units (address in bytes / 2).
+            # https://bugs.python.org/issue26647
+            self.jump_unit = 8 // oparg
+        else:
+            self.argument = struct.Struct('<H')
+            self.have_argument = dis.HAVE_ARGUMENT
+            self.jump_unit = 1
+
+    @property
+    def argument_bits(self):
+        return self.argument.size * 8
+
+
+_BYTECODE = _Bytecode()
+
+
+def _make_code(code, codestring):
+    return code.replace(co_code=codestring)
```

Comment:
Instead of manually calling `types.Code(*args)` as in [the original source code]() I am using `.replace` method and passing the updated code. This makes it work with Python > 3.8 robustly since we don't depend on the internal order of arguments to `PyCode_New`.